homeassistant/helpers/config_validation.py

```diff
@@ -446,6 +446,15 @@ def time_period_str(value: str) -> timedelta:
     elif value.startswith("+"):
         value = value[1:]
 
+    day = 0
+    date_parsed = value.partition(" ")
+    if date_parsed[1]:
+        try:
+            day = int(date_parsed[0])
+        except ValueError as err:
+            raise vol.Invalid(TIME_PERIOD_ERROR.format(value)) from err
+        value = date_parsed[2]
```

Comment:
I'm wondering, should maybe leverage `parse_duration` that we already have?

https://github.com/home-assistant/core/blob/b4791f50516dabb49ee8a0e6185d6f35af88e261/homeassistant/util/dt.py#L220

I think we are re-inventing the wheel in this method.

Reply:
Good point, didn't know that existed.

Are you saying we should change `time_period_str` into a wrapper around `parse_duration` with more discrete errors for validation sake?

Reply:
> Good point, didn't know that existed.

Didn't realize it myself before today either.

> Are you saying we should change time_period_str into a wrapper around parse_duration with more discrete errors for validation sake?

Exactly

Reply:
Updated.

Only _possible_ concern is previously `23:42` == 23 hours, 42 minutes, but using `parse_duration` makes `23:42` == 23 minutes, 42 seconds. Requires `23:42:00` format to be interpreted as 23 hours, 42 minutes.

I'm sure we can trust Django's logic here. :)